include/net/aws_iot.h

```diff
@@ -187,57 +193,52 @@ typedef void (*aws_iot_evt_handler_t)(const struct aws_iot_evt *evt);
 
 /** @brief Structure for AWS IoT broker connection parameters. */
 struct aws_iot_config {
-	/** Socket for AWS IoT broker connection */
-	int socket;
-	/** Client id for AWS IoT broker connection, used when
-	 *  @kconfig{CONFIG_AWS_IOT_CLIENT_ID_APP} is set. If not set an internal
-	 *  configurable static client id is used.
+	/** AWS IoT client ID. Must be NULL terminated.
+	 *  If not set (NULL), CONFIG_AWS_IOT_CLIENT_ID_STATIC will be used.
 	 */
 	char *client_id;
-	/** Length of client_id string. */
-	size_t client_id_len;
-	/** AWS IoT endpoint host name for broker connection, used when
-	 *  @kconfig{CONFIG_AWS_IOT_BROKER_HOST_NAME_APP} is set. If not the
-	 *  static @kconfig{AWS_IOT_BROKER_HOST_NAME} is used.
+
+	/** AWS IoT broker hostname. Must be NULL terminated.
+	 *  If not set (NULL), CONFIG_AWS_IOT_BROKER_HOST_NAME will be used.
 	 */
 	char *host_name;
-	/** Length of host_name string. */
-	size_t host_name_len;
 };
 
-/** @brief Initialize the module.
+/** @brief Initialize the library.
  *
  *  @note This API must be called exactly once, and it must return
- *        successfully.
+ *	  successfully before any other library APIs can be called.
  *
- *  @param[in] config Pointer to struct containing connection parameters.
- *  @param[in] event_handler Pointer to event handler to receive AWS IoT module
- *                           events.
+ *  @param[in] event_handler Pointer to an event handler to receive library events.
  *
- *  @return 0 If successful.
- *            Otherwise, a (negative) error code is returned.
+ *  @return 0 If successful. Otherwise, a negative error code is returned
+ *	      indicating the reason for failure.
+ *  @retval -EINVAL if an invalid parameter is passed in.
```

Comment:
I don't think it's ok to mix `@return` and `@retval` for a single function, the rendered API documentation ends up having both "Return values" and "Returns" section:
https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-11507/nrf/libraries/networking/aws_iot.html#c.aws_iot_init

Reply:
yeah, it looks a bit off. Ill update the API docs to only use retval.